src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/CertificateRequest.Load.cs

```diff
@@ -348,6 +349,20 @@ private static bool VerifyX509Signature(
                     case Oids.MLDsa87:
                         hashAlg = default;
                         break;
+                    case Oids.SlhDsaSha2_128s:
```

Comment:
Consider extracting the "Is this an SLH-DSA OID?" out to some helper and changing the switch to

```C#
case string oid when IsSlhOid(oid):
    // whatever
    break;
```

And replacing all of the large switch cases with that helper.

Reply:
Oh nice.. I was only considering breaking this case off into an if + helper after the switch but that looked too weird. The small nit I have about the switch pattern is that catch-all pattern that needs to be there to execute the `when`, but that's just a syntax complaint.